lld/MachO/Driver.cpp

```diff
@@ -407,8 +407,28 @@ static InputFile *addFile(StringRef path, LoadType loadType,
   case file_magic::macho_dynamically_linked_shared_lib_stub:
   case file_magic::tapi_file:
     if (DylibFile *dylibFile =
-            loadDylib(mbref, nullptr, /*isBundleLoader=*/false, isExplicit))
+            loadDylib(mbref, nullptr, /*isBundleLoader=*/false, isExplicit)) {
+      if (isExplicit && !dylibFile->allowableClients.empty()) {
+        bool allowed = std::any_of(
+            dylibFile->allowableClients.begin(),
+            dylibFile->allowableClients.end(), [&](StringRef allowableClient) {
+              // Not what you expect, but exactly as LD64 does.
```

Comment:
Yes, that it does a prefix match. It honestly seems like a bug, but the behaviour has been around for so long that it probably doesn't make sense to change in LD64 anymore.

Reply:
I don't object to the behavior, I just found the comment confusing as-is. Something like `// Doing a prefix match instead of an exact match matches ld64's behavior` would imho be clearer.

Reply:
Sure, I'll update the comment.

Reply:
Done now.